### PR TITLE
telemetry: replace (deprecated) pkg_resources with importlib (bug 1894839)

### PR DESCRIPTION
**Pull request by badboy**
This makes it work with Python 3.12

(or it will as soon as [I do a Glean release too](https://bugzilla.mozilla.org/show_bug.cgi?id=1894595)).
Note that I couldn't find anything how to actually manually update the dependencies; that seems undocumented but requires a push to a `requirements` branch to trigger CI for it? I will need help with that.

### CODE REVIEW

**Comment by badboy** [comment]
Apparently there's more `pkg_resource` usage in dependencies:

```
  /home/runner/work/mozregression/mozregression/env/lib/python3.8/site-packages/mozrunner/utils.py:19: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
```

so that miiiight require changes in mozrunner too.

**Comment by zzzeid** [comment]
> This makes it work with Python 3.12
> 
> (or it will as soon as [I do a Glean release too](https://bugzilla.mozilla.org/show_bug.cgi?id=1894595)). Note that I couldn't find anything how to actually manually update the dependencies; that seems undocumented but requires a push to a `requirements` branch to trigger CI for it? I will need help with that.

Thanks for this contribution! Looks like this change will be needed with the latest `setuptools`. In order to trigger a CI run to generate requirements files, the following steps can be run:
- checkout the `requirements` branch
- delete generated `*.txt` files in the `requirements` subdirectory
- commit and push
- wait for CI to finish, and pull the latest commit from that branch

That last commit will contain all the different requirements files based on any new packages that match the criteria in the `.in` files. You can then rebase and move the commit to wherever you need it.

**Comment by zzzeid** [comment]
Following up on this, there are a couple of hiccups with the Python 3.12 package upgrades, I will resolve those and hopefully once that is in place this will be easier to implement. Looks like changes are needed anyway on mozrunner as you said.

**Comment by badboy** [comment]
> Following up on this, there are a couple of hiccups with the Python 3.12 package upgrades, I will resolve those and hopefully once that is in place this will be easier to implement. Looks like changes are needed anyway on mozrunner as you said.

ah yeah. The CI failures also look ... like it might require some updates to requirements.txt.
I will leave that work to you.
This PR should be mergeable as is, it doesn't touch any deps. A new glean-sdk is being released right now, so whenever you get to the 3.12 work you should also update glean-sdk

File: mozregression/telemetry.py [hunk 1]
```diff
@@ -7,14 +7,14 @@
...
+import importlib.resources
 
 from mozregression import __version__
 from mozregression.dates import is_date_or_datetime, to_datetime
 
 LOG = get_proxy_logger("telemetry")
-PINGS = load_pings(resource_filename(__name__, "pings.yaml"))
-METRICS = load_metrics(resource_filename(__name__, "metrics.yaml"))
+PINGS = load_pings(importlib.resources.path(__name__, "pings.yaml"))
+METRICS = load_metrics(importlib.resources.path(__name__, "metrics.yaml"))
```
Comment (zzzeid) [inline mozregression/telemetry.py:17]: I think this is broken (see failing test). Perhaps it should be like this instead.

```suggestion
PINGS = load_pings(importlib.resources.files(__name__) / "pings.yaml")
METRICS = load_metrics(importlib.resources.files(__name__) / "metrics.yaml")
```

**Review by zzzeid — CHANGES_REQUESTED** [review]
Please rebase with main when you have a chance.

**Comment by zzzeid** [comment]
> Apparently there's more `pkg_resource` usage in dependencies:
> 
> ```
>   /home/runner/work/mozregression/mozregression/env/lib/python3.8/site-packages/mozrunner/utils.py:19: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
> ```
> 
> so that miiiight require changes in mozrunner too.

Looks like mozrunner [fails gracefully](https://searchfox.org/mozilla-central/source/testing/mozbase/mozrunner/mozrunner/utils.py#50) in that case.

**Comment by zzzeid** [comment]
For reasons I outlined in #1732, I think we need to use the `importlib_resources` package, at least until we drop support for Python 3.8 (but also, potentially until farther into the future). This is why tests are failing here. Feel free to copy over the implementation in my PR since you had started this change, and we can close that one. Or alternatively we can abandon this PR and I can merge the other one.

**Review by zzzeid — CHANGES_REQUESTED** [review]
Please see https://github.com/mozilla/mozregression/pull/1689#issuecomment-2168188499.

**Comment by zzzeid** [comment]
Closing this PR as it is replaced by #1689 which fixes the issues present in this PR.

**Pull request closed without merging**